.github/workflows/deploy_oidc.yml

```diff
@@ -0,0 +1,63 @@
+on:
+  workflow_call:
+    inputs:
+      deploy_env:
+        required: true
+        type: string
+    secrets:
+      aws_access_key_id:
+        required: false
+      aws_secret_access_key:
+        required: false
+      oidc_role_arn:
+        required: true
+
+jobs:
+  deploy_oidc:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.deploy_env }}
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v3
+
+      - name: Set up Python, caching dependencies
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+          cache: 'pip'
+      
+      - name: Manually install requirements.txt
+        run: pip install -r deploy/requirements.txt
+      
+      - name: Install node and related deps
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.19.1
+
+      - name: Install AWS CDK
+        run: npm install -g aws-cdk
+
+      - name: Configure awscli (local runner)
+        if: ${{ env.ACT }} # OIDC doesn't work when using ACT, as AWS user accounts don't have access to the OIDC role by default
```

Comment:
I found this a bit confusing but perhaps I don't understand something relevant about role assumption: isn't it the case that roles to be assumed depend on `trust` relationships which will be related to user or account ID? This isn't a correction so much as a curiosity about what I am missing

Reply:
As-is, the role requires a token/OIDC auth from this Github repository, so (as an example) my AWS user account running locally, without that auth from Github, cannot assume the role unless that condition is modified.